### PR TITLE
Fix compatibility with VTK 9.3.20240914

### PR DESCRIPTION
**Pull request by mwestphal**
VTK recently introduced support for dynamic loading of graphical libraries, breaking retro-compatibility in the process.

This fix compatibility with VTK while also supporting implicitely EGL/OSMESA rendering with recent versions of VTK.

`DISPLAY= ./bin/f3d` just works now.

Also updates VTK version in CI

Related to: https://github.com/f3d-app/f3d/issues/1445  https://github.com/f3d-app/f3d/issues/1621

### CODE REVIEW

File: application/CMakeLists.txt [hunk 1]
```diff
@@ -106,9 +106,11 @@ if (F3D_MODULE_EXR)
   target_compile_definitions(f3d PRIVATE F3D_MODULE_EXR)
 endif ()
 
+if(VTK_VERSION VERSION_LESS 9.3.20240914)
 # Headless EGL build
-if (VTK_OPENGL_HAS_EGL)
-  target_compile_definitions(f3d PRIVATE F3D_HEADLESS_BUILD)
+  if (VTK_OPENGL_HAS_EGL)
+    target_compile_definitions(f3d PRIVATE F3D_HEADLESS_BUILD)
```
Comment (Meakk) [inline application/CMakeLists.txt:112]: I think this variable was used to determine the render window to instantiate?
How is it working now? EGL is always available?

Comment (mwestphal) [reply application/CMakeLists.txt:112]: see discord

File: application/testing/CMakeLists.txt [hunk 2]
```diff
@@ -981,15 +986,17 @@ if(NOT F3D_MACOS_BUNDLE)
   endif()
 endif()
 
+# TODO VTK_VERSION
```
Comment (mwestphal) [inline application/testing/CMakeLists.txt:989]: remove

**Comment by Meakk** [comment]
I'm taking over with https://github.com/f3d-app/f3d/pull/1631, let's close this one.

**Pull request closed without merging**